enterprise/ha/src/main/java/org/neo4j/kernel/ha/MasterClient214.java

```diff
@@ -19,32 +19,78 @@
  */
 package org.neo4j.kernel.ha;
 
+import org.jboss.netty.buffer.ChannelBuffer;
+
+import java.io.IOException;
+
+import org.neo4j.com.Client;
+import org.neo4j.com.Deserializer;
 import org.neo4j.com.Protocol;
 import org.neo4j.com.Protocol214;
 import org.neo4j.com.ProtocolVersion;
+import org.neo4j.com.RequestContext;
+import org.neo4j.com.RequestType;
+import org.neo4j.com.Response;
+import org.neo4j.com.Serializer;
 import org.neo4j.com.monitor.RequestMonitor;
 import org.neo4j.com.storecopy.ResponseUnpacker;
+import org.neo4j.com.storecopy.StoreWriter;
+import org.neo4j.kernel.ha.com.master.HandshakeResult;
+import org.neo4j.kernel.ha.com.master.Master;
+import org.neo4j.kernel.ha.com.master.MasterServer;
+import org.neo4j.kernel.ha.com.slave.MasterClient;
+import org.neo4j.kernel.ha.id.IdAllocation;
+import org.neo4j.kernel.ha.lock.LockResult;
 import org.neo4j.kernel.impl.store.StoreId;
+import org.neo4j.kernel.impl.store.id.IdRange;
+import org.neo4j.kernel.impl.store.id.IdType;
+import org.neo4j.kernel.impl.transaction.TransactionRepresentation;
 import org.neo4j.kernel.impl.transaction.log.ReadableClosablePositionAwareChannel;
 import org.neo4j.kernel.impl.transaction.log.entry.LogEntryReader;
 import org.neo4j.kernel.monitoring.ByteCounterMonitor;
 import org.neo4j.logging.LogProvider;
+import org.neo4j.storageengine.api.lock.ResourceType;
 
+import static org.neo4j.com.Protocol.EMPTY_SERIALIZER;
+import static org.neo4j.com.Protocol.VOID_DESERIALIZER;
+import static org.neo4j.com.Protocol.writeString;
 import static org.neo4j.com.ProtocolVersion.INTERNAL_PROTOCOL_VERSION;
 
-public class MasterClient214 extends MasterClient210
+/**
+ * The {@link org.neo4j.kernel.ha.com.master.Master} a slave should use to communicate with its master. It
+ * serializes requests and sends them to the master, more specifically
+ * {@link org.neo4j.kernel.ha.com.master.MasterServer} (which delegates to
+ * {@link org.neo4j.kernel.ha.com.master.MasterImpl}
+ * on the master side.
+ */
+public class MasterClient214 extends Client<Master> implements MasterClient
 {
+    /* Version 1 first version
+     * Version 2 since 2012-01-24
+     * Version 3 since 2012-02-16
+     * Version 4 since 2012-07-05
+     * Version 5 since ?
+     * Version 6 since 2014-01-07
+     * Version 7 since 2014-03-18
+     * Version 8 since 2014-08-27
```

Comment:
A legend like this is always good to have, this could be moved to MasterClient310 (which is now the latest) and have `version 9` included as well